llarp/handlers/tun.cpp

```diff
@@ -468,6 +469,21 @@ namespace llarp
                 addr.as_array(), std::make_shared<dns::Message>(msg), isV6);
           }
         }
+        else if (ends_with(qname, ".loki"))
+        {
+          return LookupNameAsync(
```

Comment:
maybe length check here and NX if the name is too long?

Reply:
Validity rules for registration:
- not `localhost.loki`, `loki.loki`, or `snode.loki`
- if it contains no dash then max 32 characters long (assuming a leading subdomain has been stripped)
These are from general DNS requirements, and also enforced in registrations:
- Must be all [A-Za-z0-9-].  (A-Z will be lower-cased by the RPC call).
- cannot start or end with a `-`
- max 63 characters long if it does contain a dash
- cannot contain `--` in the third and fourth characters *unless* it starts with `xn--`

Reply:
added this check